src/rdkafka_admin.c

```diff
@@ -2735,3 +2837,400 @@ rd_kafka_DescribeConfigs_result_resources (
 }
 
 /**@}*/
+
+/**
+ * @name Delete Records
+ * @{
+ *
+ *
+ *
+ *
+ */
+
+static void rd_kafka_topic_partition_list_free (void *ptr) {
+        rd_kafka_topic_partition_list_destroy(ptr);
+}
+
+
+/**
+ * @brief Parse DeleteRecordsResponse and create ADMIN_RESULT op.
+ */
+static rd_kafka_resp_err_t
+rd_kafka_DeleteRecordsResponse_parse (rd_kafka_op_t *rko_req,
+                                      rd_kafka_op_t **rko_resultp,
+                                      rd_kafka_buf_t *reply,
+                                      char *errstr, size_t errstr_size) {
+        const int log_decode_errors = LOG_ERR;
+        rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR_NO_ERROR;
+        rd_kafka_broker_t *rkb = reply->rkbuf_rkb;
+        rd_kafka_t *rk = rkb->rkb_rk;
+        rd_kafka_op_t *rko_result = NULL;
+        int32_t topic_cnt;
+        int i;
+        int32_t Throttle_Time;
+
+        rd_kafka_AdminFanoutState_t *state =
+                rd_kafka_confval_get_ptr(&rko_req->rko_u.admin_request.options.opaque);
+        rd_kafka_topic_partition_list_t *offsets = rd_list_elem(&state->responses, 0);
+
+        /* throttle_time_ms */
+        rd_kafka_buf_read_i32(reply, &Throttle_Time);
+        rd_kafka_op_throttle_time(rkb, rk->rk_rep, Throttle_Time);
+
+        /* #topics */
+        rd_kafka_buf_read_i32(reply, &topic_cnt);
+
+        for (i = 0 ; i < (int)topic_cnt ; i++) {
+                rd_kafkap_str_t ktopic;
+                int32_t partition_cnt;
+                int j;
+
+                /* topic */
+                rd_kafka_buf_read_str(reply, &ktopic);
+
+                /* #partitions */
+                rd_kafka_buf_read_i32(reply, &partition_cnt);
+
+                for (j = 0 ; j < (int)partition_cnt ; j++) {
+                        int32_t partition;
+                        rd_kafka_topic_partition_t *skel;
+
+                        /* partition */
+                        rd_kafka_buf_read_i32(reply, &partition);
+
+                        skel = rd_kafka_topic_partition_list_add(offsets,
+                                                                 ktopic.str,
+                                                                 partition);
+                        /* low_watermark */
+                        rd_kafka_buf_read_i64(reply, &skel->offset);
+
+                        /* error_code */
+                        rd_kafka_buf_read_i16(reply, &skel->err);
+                }
+        }
+
+        if (--state->outstanding == 0) {
+                rd_kafka_op_t *rko_result = rd_kafka_admin_result_new(rko_req);
```

Comment:
This shadows the parent scope variable.

Also, suggest reversing this if-clause to cut down on sub-scoping, as so:

```c
if (--state->outstanding > 0) {
    /* still.. */
   return ..PARTIAL;
}

rko_result = ...;
...

return ..NO_ERROR;
```